lib/runtime/procs/fanin.js

```diff
@@ -130,11 +131,22 @@ class Input {
 
         if (timelessInput) { return timelessInput; }
 
+        function hasInterval(a) {
+            return a.mark && a.interval;
+        }
+
         return _.reduce(pendingInputs, (earliest, input) => {
             var inPt = input.first();
             var earliestPt = earliest.first();
 
             if (inPt.time.lt(earliestPt.time) || (inPt.mark && inPt.time.eq(earliestPt.time))) {
+                if (hasInterval(inPt) && hasInterval(earliestPt)) {
```

Comment:
It would be good to add a comment explaining why is this done and a test verifying the behavior.